src/hpv_query.py

```diff
@@ -83,63 +84,55 @@ def process_one_file(file_path: str, text_key: str = "content") -> Iterable[dict
 def process_all_files(
     file_paths: list,
     save_file: pathlib.Path,
-    log_file: pathlib.Path,
     text_key="content",
 ):
     """
-    Process all files with an option to overwrite existing results and automatically reprocess the last file in the log if it exists.
-
-    Reprocessing the last file ensures that no data is complete if the process was interrupted mid-file.
+    Process all files and save the results in a jsonl file.
 
     args:
         file_paths: list
         save_file: pathlib.Path
-        log_file: pathlib.Path
         text_key: str
     """
-    for file in [save_file, log_file]:
-        if (
-            file.exists()
-        ):  # since we are appending to the logfile + savefile, we want to remove it if we re-run the script (to avoid duplicates)
-            file.unlink()
+    if save_file.exists() and save_file.suffix == ".jsonl": 
+        save_file.unlink() # since we are appending to the savefile, we want to remove it if we re-run the script (to avoid duplicates)
 
     for file_path in tqdm(file_paths):
         queried_entries = process_one_file(file_path, text_key)
 
-        article_count = 0
         with open(save_file, "a") as f:
             for entry in queried_entries:
                 f.write(json.dumps(entry) + "\n")
-                article_count += 1
-
-        with open(log_file, "a") as f:
-            log_data = {"file_path": str(file_path), "hpv_article_count": article_count}
-            f.write(json.dumps(log_data) + "\n")
 
+def input_parse():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
```

Comment:
Maybe wanna add a default value here? I also think that by default you probably want to set the location relative to the data folder, right?
So maybe something like:

```python
parser.add_argument("--save_path", default=None, ...)
args = parser.parse_args()

if args.save_path is None:
     save_path = corpus_path.parent / "hpv_data.jsonl"
else:
     save_path = Path(args.save_path)
```

We could also make it a bit clearer that this is a directory that should be passed, not a filename?
Or even better, why not make the user specify the file path if we are only saving a single file?
And then set some useful default.

Another little thing, though feel free to ignore: It's usually convention in UnixLand to use `-o` as an output argument when your CLI outputs a single file and then maybe name the argument something like `--out_file` to make it absolutely clear that it is a file we're talking about.

Feel free to use directory as input argument if you see fit :D

Reply:
I love conventions haha - was not aware of this one, but I'll defo make that change. The reason for not passing a default argument was actually to make it more difficult for the user to overwrite the file (i.e., by having to specify a directory themselves, they may be more conscious of what they are doing). But we can definitely do a default path here.

Reply:
:D